src/js/core/widget/core/Spin.js

```diff
@@ -0,0 +1,630 @@
+						} else if (value < self.options.min) {
+							value = self.options.max;
+						}
+					}
+					self.options.value = value;
+					// set data-value on element
+					self.element.dataset.value = value;
+
+					// stop previous animation
+					animation = self.state.animation[0];
+					if (animation !== null && animation.to !== animation.current) {
+						self._animation.stop();
+					}
+					// update status of widget
+					self._show(enableChangeEvent);
+				}
+			}
+		};
+
+		prototype._getValue = function () {
+			return this.options.value;
+		};
+
+		prototype._setMax = function (element, max) {
+			var options = this.options;
+
+			options.max = (max !== undefined) ? parseInt(max, 10) : 0;
+			this.length = options.max - options.min + 1;
+		};
+
+		prototype._setMin = function (element, min) {
+			var options = this.options;
+
+			options.min = (min !== undefined) ? parseInt(min, 10) : 0;
+			this.length = options.max - options.min + 1;
+		};
+
+		prototype._setLabels = function (element, value) {
+			var self = this;
+
+			self.options.labels = value.split(",");
+			self._refresh();
+		};
+
+		prototype._setModuloValue = function (element, value) {
+			this.options.moduloValue = (value === "enabled") ? "enabled" : "disabled";
+		};
+
+		prototype._setShortPath = function (element, value) {
+			this.options.shortPath = (value === "enabled") ? "enabled" : "disabled";
+		};
+
+		prototype._setLoop = function (element, value) {
+			this.options.loop = (value === "enabled") ? "enabled" : "disabled";
+		};
+
+		prototype._setDuration = function (element, value) {
+			this.options.duration = window.parseInt(value, 10);
+		};
+
+		prototype._setEnabled = function (element, value) {
+			var self = this;
+
+			self.options.enabled = (value === "false") ? false : value;
+			if (self.options.enabled) {
+				element.classList.add(classes.ENABLING);
+				window.setTimeout(function () {
+					element.classList.remove(classes.ENABLING);
+				}, ENABLING_DURATION);
+				element.classList.add(classes.ENABLED);
+				utilsEvents.on(document, "drag dragend", self);
+
+				// disable tau rotaryScroller the widget has own support for rotary event
+				ns.util.rotaryScrolling && ns.util.rotaryScrolling.lock();
+			} else {
+				element.classList.add(classes.ENABLING);
+				window.setTimeout(function () {
+					element.classList.remove(classes.ENABLING);
+					self.refresh();
+				}, ENABLING_DURATION);
+				element.classList.remove(classes.ENABLED);
+				utilsEvents.off(document, "drag dragend", self);
+				// disable animation
+				self._animation.stop();
+				// enable tau rotaryScroller the widget has own support for rotary event
+				ns.util.rotaryScrolling && ns.util.rotaryScrolling.unlock();
+			}
+			// reset previous value;
+			this._prevValue = null;
+			return true;
+		};
+
+		prototype._setDirection = function (element, direction) {
+			this.options.direction = (["up", "down"].indexOf(direction) > -1) ?
+				direction : "up";
```

Comment:
In my opinion, merging lines into one does not seem long.

Reply:
Okay I will check it. I just copy it from the origin code :)